beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -91,6 +94,10 @@ func (s *Service) validateBeaconBlockPubSub(ctx context.Context, pid peer.ID, ms
 		}()
 	}
 
+	if err := validateDenebBeaconBlock(blk.Block()); err != nil {
+		return pubsub.ValidationReject, nil
+	}
+
```

Comment:
Aren't you verifying twice? It's already in `validateBeaconBlock`

Reply:
I was confused about the difference between these two methods, will look again.

Reply:
Oh I see my mistake. Yes I will remove it from the other func.

Reply:
I get what's going on with these validation funcs now. If the parent of a block is known, `validateBeaconBlockPubSub` calls `validateBeaconBlock`. Otherwise the block gets stuck in the pending queue, and later, once the parent is known, `validateBeaconBlock` is finally called. In other words, `validateBeaconBlock` could be named `validateBeaconBlockWithKnownParent` or something - it's job is to keep separate logic that depends on the block's parent. 

Since this check can happen before most of the other checks, I'll only call it from `validateBeaconBlockPubSub` so it can fail as fast as possible.

Reply:
I think it's better to have it in `validateBeaconBlock` because that's also used during the pending queue. We do want this check for blocks requested over the RPC by root method

Reply:
In that case, maybe the double validation is ok? It's very cheap and it avoids using resources on a block that we could quickly reject at pub/sub time.

Reply:
Why not just have it in `validateBeaconBlock`? We only allow one block per proposer anyway, so I'm not concerned about spamming. The spec does not give any recommendation here, but I get the sentiment to have this earlier because it's cheap. The downside is it's a bit confusing to the reader, and having to explain later this is not a bug, it's a feature. I am not going to die on either side of the hill. Will let you decide. But if we do want to double checks, I reckon we add a comment for it